brave/brave-core
# Unblind token fix

Resolves https://github.com/brave/brave-browser/issues/8207
Resolves https://github.com/brave/brave-browser/issues/8180

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
**Test plan 1:**
defined in the issue

**Test plan 2:**
1. start browser pointing to dev
2. claim grant (write down promotion id)
3. close browser
4. edit Preferences file and change `promotion_corrupted_migrated` to `false`
5. open ` publisher_info_db` file and go to `promotion_creds` table. Modify `blidedn_creds` value. Replace first string with `0tB84zwByAdoiIpdV8dcifEyLUTzMNONHCLDO4Uzkm0=`
6. start browser again
7. make sure that `promotion` and `promotion_creds` is empty
8. make sure that you see request going out with your promotion id looking something like this
```
[ REQUEST ]
> url: https://grant.rewards.brave.software/v1/promotions/reportclobberedclaims
> method: UrlMethod::POST
> content: {"claimIds":["811620a6-b4ed-416d-99c4-d29e0a1ddf08"]}
> contentType: application/json; charset=utf-8
[ END REQUEST ]
```

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
